1-Authentication/1-sign-in/README-incremental.md

```diff
@@ -0,0 +1,273 @@
+
+You may also use MSAL React's [useMsalAuthentication](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/hooks.md#usemsalauthentication-hook) hook. Below is an example in which the `ssoSilent()` API is used. When using `ssoSilent()`, the recommended pattern is that you fallback to an **interactive method** should the silent SSO attempt fails:
+
+```javascript
+function App() {
+    const request = {
+        loginHint: "name@example.com",
+        scopes: ["User.Read"]
+    }
+
+    const { login, result, error } = useMsalAuthentication(InteractionType.Silent, request);
+
+    useEffect(() => {
+        if (error) {
+            login(InteractionType.Popup, request);
+        }
+    }, [error]);
+
+    const { accounts } = useMsal();
+
+    return (
+        <React.Fragment>
+            <p>Anyone can see this paragraph.</p>
+            <AuthenticatedTemplate>
+                <p>Signed in as: {accounts[0]?.username}</p>
+            </AuthenticatedTemplate>
+            <UnauthenticatedTemplate>
+                <p>No users are signed in!</p>
+            </UnauthenticatedTemplate>
+        </React.Fragment>
+    );
+}
+```
+
+As shown above, the components that depend on whether the user is authenticated should be wrapped inside React's `AuthenticatedTemplate` and `UnauthenticatedTemplate` components. Alternatively, you may use the [useIsAuthenticated](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/getting-started.md#useisauthenticated-hook) hook to conditionally render components.
+
+### Sign-out
+
+The application redirects the user to the **Microsoft identity platform** logout endpoint to sign out. This endpoint clears the user's session from the browser. If your app does not go to the logout endpoint, the user may re-authenticate to your app without entering their credentials again, because they would have a valid single sign-on session with the **Microsoft identity platform** endpoint. For more, see: [Send a sign-out request](https://docs.microsoft.com/azure/active-directory/develop/v2-protocols-oidc#send-a-sign-out-request)
+
+### ID token validation
+
+A single-page application does not benefit from validating [ID tokens](https://docs.microsoft.com/azure/active-directory/develop/id-tokens), since the application runs without a back-end and as such, attackers can intercept and edit the keys used for validation of the token.
+
+### Sign-in audience and account types
+
+This sample is meant to work with accounts in your organization (aka *single-tenant*). If you would like to allow sign-ins from other organizations and/or with other types of accounts, you have to configure your `authority` string in `authConfig.js` accordingly. For example:
+
+```javascript
+const msalConfig = {
+    auth: {
+      clientId: "<YOUR_CLIENT_ID>",
+      authority: "https://login.microsoftonline.com/consumers", // allows sign-ins with personal Microsoft accounts.
```

Comment:
>authority: "https://login.microsoftonline.com/consumers", // allows sign-ins with personal Microsoft accounts. [](start = 6, length = 110)

why 'consumers' as example here?

Reply:
just an example, I've put a link to see other audience options.

Reply:
MSAL is a chapter in itself, think about what issues they will run into if they copied and pasted from here

---
In reply to: [577881386](https://github.com/Azure-Samples/ms-identity-javascript-react-tutorial/pull/1#discussion_r577881386) [](ancestors = 577881386)